### ISSUE TITLE
aiven_transit_gateway_vpc_attachment with user_peer_network_cidrs makes provider crash

### ISSUE THREAD

**Issue by jeffrichie** [aiven/terraform-provider-aiven#532]
Does not crash:

```
resource "aiven_transit_gateway_vpc_attachment" "aiven_vpn_attachment" {
    vpc_id = aiven_project_vpc.hf_aiven_vpc.id
    peer_cloud_account = data.aws_caller_identity.current.account_id
    peer_vpc = aws_ec2_transit_gateway.aiven_transit_gateway.id
    peer_region = var.region
    user_peer_network_cidrs = []
```

Crashes:

```
resource "aiven_transit_gateway_vpc_attachment" "aiven_vpn_attachment" {
    vpc_id = aiven_project_vpc.hf_aiven_vpc.id
    peer_cloud_account = data.aws_caller_identity.current.account_id
    peer_vpc = aws_ec2_transit_gateway.aiven_transit_gateway.id
    peer_region = var.region
    user_peer_network_cidrs = [ 
        "10.0.0.0/16"
    ]
```

Crash message:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # module.aiven_vpc.aiven_transit_gateway_vpc_attachment.aiven_vpn_attachment will be updated in-place
  ~ resource "aiven_transit_gateway_vpc_attachment" "aiven_vpn_attachment" {
        id                      = "nacelle-shared-infra-development/00bb9571-1b45-404c-a0b3-26128814af2b/487200645058/tgw-076b5352db0f45e5d/us-east-2"
      ~ user_peer_network_cidrs = [
          + "10.0.0.0/16",
        ]
        # (6 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
module.aiven_vpc.aiven_transit_gateway_vpc_attachment.aiven_vpn_attachment: Modifying... [id=nacelle-shared-infra-development/00bb9571-1b45-404c-a0b3-26128814af2b/487200645058/tgw-076b5352db0f45e5d/us-east-2]

Stack trace from the terraform-provider-aiven_v2.1.17 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1323fed]

goroutine 28 [running]:
github.com/aiven/aiven-go-client.(*TransitGatewayVPCAttachmentHandler).Update(0x0, 0xc000139a00, 0x20, 0xc000139a21, 0x24, 0xc0000b64c0, 0x1, 0x1, 0x0, 0x0, ...)
	github.com/aiven/aiven-go-client@v1.5.12-0.20210701142609-bc899bc1da90/transit_gateway_vpc_attachment.go:32 +0x16d
github.com/aiven/terraform-provider-aiven/aiven.resourceTransitGatewayVPCAttachmentUpdate(0x1d8f598, 0xc00068c300, 0xc0000b5980, 0x1b48720, 0xc000624fc0, 0xc0007454b0, 0x146966a, 0xc0009d1b00)
	github.com/aiven/terraform-provider-aiven/aiven/resource_transit_gateway_vpc_attachement.go:130 +0x63e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00038e2a0, 0x1d8f528, 0xc000575600, 0xc0000b5980, 0x1b48720, 0xc000624fc0, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:364 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00038e2a0, 0x1d8f528, 0xc000575600, 0xc000a49340, 0xc0009d1b00, 0x1b48720, 0xc000624fc0, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0000e04c8, 0x1d8f528, 0xc000575600, 0xc0004545a0, 0xc000575600, 0x1b6dda0, 0xc0004ff700)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0009d0f20, 0x1d8f5d0, 0xc000575600, 0xc000a49180, 0xc0009d0f20, 0xc0004ff7a0, 0xc000705ba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1b6dda0, 0xc0009d0f20, 0x1d8f5d0, 0xc0004ff7a0, 0xc000a255c0, 0x0, 0x1d8f5d0, 0xc0004ff7a0, 0xc000273800, 0x5e9)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00036fdc0, 0x1d98438, 0xc00050ef00, 0xc0000ba700, 0xc000888510, 0x22c88c0, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00036fdc0, 0x1d98438, 0xc00050ef00, 0xc0000ba700, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0004cc180, 0xc00036fdc0, 0x1d98438, 0xc00050ef00, 0xc0000ba700)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-aiven_v2.1.17 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.


Error: Plugin did not respond

  with module.aiven_vpc.aiven_transit_gateway_vpc_attachment.aiven_vpn_attachment,
  on modules/aiven-vpc/main.tf line 45, in resource "aiven_transit_gateway_vpc_attachment" "aiven_vpn_attachment":
  45: resource "aiven_transit_gateway_vpc_attachment" "aiven_vpn_attachment" {

The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
```

**Comment by mhoffm-aiven** [comment]
Fixed upstream by https://github.com/aiven/aiven-go-client/pull/122

**Comment by mhoffm-aiven** [comment]
Should be fixed in the v1.5.18
